xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/java/org/xwiki/model/validation/edit/internal/DefaultEditConfirmationCheckersManager.java

```diff
@@ -0,0 +1,157 @@
+import org.xwiki.component.manager.ComponentLookupException;
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.model.reference.EntityReferenceSerializer;
+import org.xwiki.model.validation.edit.EditConfirmationChecker;
+import org.xwiki.model.validation.edit.EditConfirmationCheckerResult;
+import org.xwiki.model.validation.edit.EditConfirmationCheckerResults;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+
+import static org.apache.commons.lang3.exception.ExceptionUtils.getRootCauseMessage;
+
+/**
+ * Default implementation of {@link EditConfirmationCheckersManager}.
+ * Stores the forced results in the user's HTTP session.
+ *
+ * @version $Id$
+ * @since 15.10RC1
+ */
+@Component
+@Singleton
+public class DefaultEditConfirmationCheckersManager implements EditConfirmationCheckersManager
+{
+    private static final String CACHED_QUALIFIER = "cached";
+
+    private static final String FORCED_QUALIFIER = "forced";
+
+    @Inject
+    @Named("context")
+    private ComponentManager componentManager;
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private EntityReferenceSerializer<String> entityReferenceSerializer;
+
+    @Inject
+    private Provider<XWikiContext> xcontextProvider;
+
+    @Override
+    public EditConfirmationCheckerResults check()
+    {
+        EditConfirmationCheckerResults result = new EditConfirmationCheckerResults();
+        XWikiDocument xWikiDocument = this.xcontextProvider.get().getDoc();
+        String cachedKeyPrefix = computePrefix(xWikiDocument, CACHED_QUALIFIER);
+        String forcedKeyPrefix = computePrefix(xWikiDocument, FORCED_QUALIFIER);
+
+        HttpSession session = getSession();
+
+        // The list is ordered by the priority of the components.
+        confirmationCheckersStream()
+            .flatMap(entry -> {
+                String forcedKey = computeKey(entry, forcedKeyPrefix);
+                String cachedKey = computeKey(entry, cachedKeyPrefix);
+                Object forcedValue = session.getAttribute(forcedKey);
+                EditConfirmationChecker checker = entry.getValue();
+                Optional<EditConfirmationCheckerResult> check = checker.check();
+                check.ifPresentOrElse(editConfirmationCheckerResult -> {
+                    if (!result.isError()) {
+                        session.setAttribute(cachedKey, editConfirmationCheckerResult.getCache());
+                    }
+                }, () -> session.removeAttribute(cachedKey));
+                Optional<EditConfirmationCheckerResult> checkResult = check
+                    .flatMap(r -> {
+                        // We ignore the result if this is not an error, the cache value is not null, and the  
+                        // cache value stored in session is equal to the cache value returned by the analysis.
+                        if (!r.isError() && forcedValue != null && Objects.equals(r.getCache(), forcedValue)) {
```

Comment:
So my understanding is that this `getCache` value is actually not something you want to have in cache, but something you want to use for *invalidating* the cache, is that correct? I'm not sure the name properly reflect that.

Reply:
`cache` renamed in `skipValue` wdyt?